code/vocabulary.py

```diff
@@ -90,3 +97,86 @@
 # for quick verification of the reload
 # print(str(ctx.settings["dictate.word_map"]))
 # print(str(ctx.lists["user.vocabulary"]))
+
+
+class PhraseReplacer:
+    """Utility for replacing phrases by other phrases inside text or word lists.
+
+    Replacing longer phrases has priority.
+
+    Args:
+      - phrase_dict: dictionary mapping recognized/spoken forms to written forms
+    """
+
+    def __init__(self, phrase_dict: Dict[str, str]):
+        # Index phrases by first word, then number of subsequent words n_next
+        phrase_index = dict()
+        for spoken_form, written_form in phrase_dict.items():
+            words = spoken_form.split()
+            first_word, n_next = words[0], len(words) - 1
```

Comment:
You might want to handle the case that `spoken_form.split()` is an empty list here, probably by logging a warning (`logging.warning("blah")`) and ignoring that item. Otherwise if a user screws up their csv file by adding an empty spoken form, all replacements will fail to work.

Reply:
Good point, done!